### ISSUE TITLE
Pagination

### ISSUE THREAD

**Issue by thiagogabriel** [inaka/rest_guidelines#7]
Since we should return a plain array with the items, where would we include the pagination metadata ?

The practice I know is do something like this:

```json
{ "items" :
    [
        {
            "like_count": 0,
            "id": "string",
            "media_type": "comment",
            "owner": "string"
        },
        {
            "like_count": 0,
            "id": "string",
            "media_type": "comment",
            "owner": "string"
        },
        {
            "like_count": 0,
            "id": "string",
            "media_type": "comment",
            "owner": "string"
        }
    ],
  "meta":
    {
        "current_page": 1,
        "next_page": null,
        "total_pages": 1,
        "prev_page": null
    }
}
```

**Comment by agerace** [comment]
I'll have to check with @elbrujohalcon  how we usually do this, but I think it depends on the project. Sometimes we use page numbers, sometimes we use a cursor. As far as I know, the paginated resources are the only ones that should get enveloped items, as you wrote above @thiagogabriel . I'm gonna check if there's no item on the momentary index for this and, if there isn't, add it .

**Comment by elbrujohalcon** [comment]
Pagination can be in the headers, right @volbap ?

**Comment by elbrujohalcon** [comment]
In any case, we can totally add pagination as an exception to that rule, if needed.

**Comment by volbap** [comment]
Yeah, I would use the link headers as explained [here](http://www.vinaysahni.com/best-practices-for-a-pragmatic-restful-api#pagination). It seems to be a fair solution.

Also, let's try to [avoid envelopes](http://www.vinaysahni.com/best-practices-for-a-pragmatic-restful-api#envelope) as well.

**Comment by volbap** [comment]
What [Jayme](https://github.com/inaka/Jayme/) currently supports is a non-cursor based pagination, using these headers:
 `X-Total`, `X-Per-Page`, `X-Page`

This kind of pagination, however, may not be quite useful for certain common scenarios, like a scroll feed. So, I would not make it a standard.
